From 2fe801b96661be1c09ab786c8f62394746e971de Mon Sep 17 00:00:00 2001
From: thongsyho <thong.ho.px@rvc.renesas.com>
Date: Wed, 4 Apr 2018 14:51:56 +0700
Subject: [PATCH 492/628] ARM: DTS: r8a7744: Add Video Processing Unit Cache
 (VPC module) and multi-codec module (VCP module) node to device tree

This patch support Renesas UVCS kernel module (VPC module and VCP module) probe.

Signed-off-by: thongsyho <thong.ho.px@rvc.renesas.com>
Signed-off-by: vietn <vietn@fsoft.com.vn>
---
 arch/arm/boot/dts/r8a7742.dtsi | 58 ++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 58 insertions(+)

diff --git a/arch/arm/boot/dts/r8a7742.dtsi b/arch/arm/boot/dts/r8a7742.dtsi
index 81b752a..b8773b1 100644
--- a/arch/arm/boot/dts/r8a7742.dtsi
+++ b/arch/arm/boot/dts/r8a7742.dtsi
@@ -685,6 +685,64 @@
 		power-domains = <&sysc R8A7742_PD_ALWAYS_ON>;
 	};
 
+	vcp0: vcp0@fe900000 {
+		compatible = "renesas,vcp0";
+		reg = <0 0xfe900000 0 0x200>, <0 0xfe900200 0 0x200>;
+		interrupts = <0 258 IRQ_TYPE_LEVEL_HIGH>,
+			 <0 259 IRQ_TYPE_LEVEL_HIGH>;
+		clocks = <&mstp1_clks R8A7742_CLK_VCP0>;
+		power-domains = <&sysc R8A7742_PD_ALWAYS_ON>;
+		renesas,#ch = <0>;
+	};
+
+	vpc0: vcp0@fe908000 {
+		compatible = "renesas,vpc0";
+		reg = <0 0xfe908000 0 0x90>;
+		clocks = <&mstp1_clks R8A7742_CLK_VPC0>;
+		power-domains = <&sysc R8A7742_PD_ALWAYS_ON>;
+		renesas,#ch = <0>;
+	};
+
+	vcp1: vcp1@fe900000 {
+		compatible = "renesas,vcp1";
+		reg = <0 0xfe910000 0 0x200>, <0 0xfe910200 0 0x200>;
+		interrupts = <0 260 IRQ_TYPE_LEVEL_HIGH>,
+			 <0 261 IRQ_TYPE_LEVEL_HIGH>;
+		clocks = <&mstp1_clks R8A7742_CLK_VCP1>;
+		power-domains = <&sysc R8A7742_PD_ALWAYS_ON>;
+		renesas,#ch = <1>;
+	};
+
+	vpc1: vcp0@fe918000 {
+		compatible = "renesas,vpc1";
+		reg = <0 0xfe918000 0 0x90>;
+		clocks = <&mstp1_clks R8A7742_CLK_VPC1>;
+		power-domains = <&sysc R8A7742_PD_ALWAYS_ON>;
+		renesas,#ch = <1>;
+	};
+
+	vpc0xy: vcp0@fe960380 {
+		compatible = "renesas,vpc0xy";
+		reg = <0 0xfe960380 0 0x4>;
+		clocks = <&mstp1_clks R8A7742_CLK_VPC0>;
+		power-domains = <&sysc R8A7742_PD_ALWAYS_ON>;
+		renesas,#ch = <0>;
+	};
+
+	vpc1xy: vcp0@fe960384 {
+		compatible = "renesas,vpc1xy";
+		reg = <0 0xfe960384 0 0x4>;
+		#clocks = <&mstp1_clks R8A7742_CLK_VPC1>;
+		power-domains = <&sysc R8A7742_PD_ALWAYS_ON>;
+		renesas,#ch = <1>;
+	};
+
+	prr: vcp0@ff000044 {
+		compatible = "renesas,prr";
+		reg = <0 0xff000044 0 0x4>;
+		renesas,#ch = <0>;
+	};
+
 	vspd0: vspd0@fe930000 {
 		compatible = "renesas,vsp2";
 		reg = <0 0xfe930000 0 0x7404>;
-- 
2.7.4

